pravega/src/test/java/docs/javadsl/PravegaGraphTestCase.java

```diff
@@ -0,0 +1,104 @@
+import akka.stream.alpakka.pravega.WriterSettingsBuilder;
+import akka.stream.javadsl.Source;
+import com.google.common.util.concurrent.Futures;
+import io.pravega.client.stream.impl.JavaSerializer;
+import org.junit.Assert;
+import org.junit.Test;
+
+import akka.japi.Pair;
+
+import akka.stream.UniqueKillSwitch;
+import akka.stream.KillSwitches;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.alpakka.pravega.javadsl.Pravega;
+
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class PravegaGraphTestCase extends PravegaBaseTestCase {
+
+  // #configuration
+  private WriterSettings<String> writerSettings =
+      WriterSettingsBuilder.create(system).withSerializer(new JavaSerializer<>());
+
+  private WriterSettings<String> writerSettingsWithRoutingKey =
+      WriterSettingsBuilder.create(system)
+          .withSerializer(new JavaSerializer<>(), (String str) -> str.substring(0, 2));
+
+  private ReaderSettings<String> readerSettings =
+      ReaderSettingsBuilder.create(system).withSerializer(new JavaSerializer<>());
+  // #configuration
+
+  @Test
+  public void infiniteSourceTest()
+      throws ExecutionException, InterruptedException, TimeoutException {
+
+    // #writing
+    Sink<String, CompletionStage<Done>> sinkWithRouting =
+        Pravega.sink(scope, streamName, writerSettings);
+
+    CompletionStage<Done> doneWithRouting =
+        Source.from(Arrays.asList("One", "Two", "Three"))
+            .toMat(sinkWithRouting, Keep.right())
+            .run(materializer);
+
+    // #writing
+
+    Sink<String, CompletionStage<Done>> sink =
+        Pravega.sink(scope, streamName, writerSettingsWithRoutingKey);
+
+    CompletionStage<Done> done =
+        Source.from(Arrays.asList("One", "Two", "Three"))
+            .toMat(sink, Keep.right())
+            .run(materializer);
+
+    CompletableFuture.allOf(done.toCompletableFuture(), doneWithRouting.toCompletableFuture())
+        .get();
+
+    CompletableFuture<Boolean> countTo200 = new CompletableFuture<>();
+    // #reading
+    Pair<UniqueKillSwitch, CompletionStage<Integer>> pair =
+        Pravega.source(scope, streamName, readerSettings)
+            .map(e -> e.message())
+            .viaMat(KillSwitches.single(), Keep.right())
+            .toMat(
+                Sink.fold(
+                    0,
+                    new Function2<Integer, String, Integer>() {
+                      @Override
+                      public Integer apply(Integer acc, String str) throws Exception, Exception {
+                        if (acc == 5) countTo200.complete(true);
+                        return acc + 1;
+                      }
+                    }),
+                Keep.both())
+            .run(materializer);
+    // #reading
```

Comment:
I found this example unnecessary complex when reading the docs. I mean, the docs expect a small snippet on how to create a `Source` and this code is implementing a full test. I like the idea to reuse test code in the docs, but only if we can refactor the code such that the docs are meaningful.

Reply:
It's definitely an involved example.  I think it could be simplified.